packages/kbn-unified-field-list/src/components/field_stats/field_stats.tsx

```diff
@@ -187,6 +191,18 @@ const FieldStatsComponent: React.FC<FieldStatsProps> = ({
       abortControllerRef.current?.abort();
       abortControllerRef.current = new AbortController();
 
+      /**
+       * If the ES|QL query is unsupported, we can exit early
+       */
+      const unsupportedReasonForQuery = isTextBased
+        ? getReasonIfFieldStatsUnavailableForQuery(query, FIELD_DATA_LABEL)
+        : undefined;
```

Comment:
This file is quite large already. Let's move this logic to `fetchAndCalculateFieldStats` handler in https://github.com/elastic/kibana/blob/b6287708f687d4e3288851052c0c6ae4ade8ce60/packages/kbn-unified-field-list/src/services/field_stats_text_based/field_stats_utils_text_based.ts#L56

Reply:
I ended up refactoring it to `loadFieldStatsTextBased` instead here [`4215c9c` (#197538)](https://github.com/elastic/kibana/pull/197538/commits/4215c9c5c7a50cf6d4970f44c3e506846b6aa76d)